ucp/_libs/utils.pyx

```diff
@@ -16,23 +18,17 @@ def get_buffer_data(buffer, check_writable=False):
     Returns data pointer of the buffer. Raising ValueError if the buffer
     is read only and check_writable=True is set.
     """
-    iface = None
-    if hasattr(buffer, "__cuda_array_interface__"):
-        iface = buffer.__cuda_array_interface__
-    elif hasattr(buffer, "__array_interface__"):
-        iface = buffer.__array_interface__
+
+    iface = getattr(buffer, "__array_interface__", None)
+    if iface is None:
+        iface = getattr(buffer, "__cuda_array_interface__", None)
 
     if iface is not None:
-        data_ptr, data_readonly = iface['data']
+        data_ptr, data_readonly = iface["data"]
     else:
         mview = memoryview(buffer)
-        data_ptr = int(<uintptr_t>PyMemoryView_GET_BUFFER(mview).buf)
-        data_readonly = mview.readonly
-
-    # Workaround for numba giving None, rather than an 0.
-    # https://github.com/cupy/cupy/issues/2104 for more info.
-    if data_ptr is None:
-        data_ptr = 0
+        data_ptr = <uintptr_t>PyMemoryView_GET_BUFFER(mview).buf
+        data_readonly = <bint>PyMemoryView_GET_BUFFER(mview).readonly
```

Comment:
Can store an intermediate of the result of `PyMemoryView_GET_BUFFER(mview)` here to save a call.

Reply:
[`PyMemoryView_GET_BUFFER`]( https://github.com/python/cpython/blob/d905df766c367c350f20c46ccd99d4da19ed57d8/Include/memoryobject.h#L18 ) is just a macro that access the `Py_buffer` object owned by the `memoryview`.